frontend/my-app/src/components/BarChartHorizontalDashboard.js

```diff
@@ -0,0 +1,80 @@
+import React, {useState, useEffect} from 'react'
+import {Chart as ChartJS, BarElement, CategoryScale, LinearScale} from 'chart.js'
+import {Bar} from 'react-chartjs-2'
+
+ChartJS.register(
+    CategoryScale,
+    LinearScale,
+    BarElement
+)
+
+const BarChartHorizontalDashboard = () => {
+    const [chart, setChart] = useState([]);
+
+    useEffect(() => {
+     const fetchData = async () => {
+        await fetch('http://localhost:5000/meetingHistory', {
+            method: 'GET',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+        }).then((response) => {
+            response.json().then((json) => {
+                //find last meeting number for each
+                var lastGen = 2;
+                var lastEng = 2;
+                var lastDes = 2;
+                var lastProd = 2;
+                var gendata = json.filter(x => (x.meetingType === 'Product' && x.meetingNumber == lastProd)||(x.meetingType === 'Engineering' && x.meetingNumber == lastEng) || (x.meetingType === 'General' && x.meetingNumber == lastGen) || (x.meetingType === 'Design' && x.meetingNumber == lastDes));
```

Comment:
So just a thought. You have a backend function (/meetingHistory) where you're calling for many cases and yet you're doing all the filtering/sorting logic on the frontend. Could you instead pass in what data you want to the backend and do the sorting there so you have minimal processing on the frontend? This way, you're not sending so much data to the frontend to process, leading to faster loading times.